bundles/core/org.eclipse.smarthome.core.ephemeris/src/main/java/org/eclipse/smarthome/core/ephemeris/internal/EphemerisManagerImpl.java

```diff
@@ -0,0 +1,237 @@
+    private final static Logger logger = LoggerFactory.getLogger(EphemerisManagerImpl.class);
+
+    // constants for the configuration properties
+    protected static final String CONFIG_URI = "system:ephemeris";
+    private static final String CONFIG_WEEKENDAYS = "weekendDays";
+    private static final String CONFIG_COUNTRY = "country";
+    private static final String CONFIG_REGION = "region";
+    private static final String CONFIG_CITY = "city";
+
+    private final Set<DayOfWeek> weekend = new HashSet<>();
+    private final HashMap<Object, HolidayManager> holidayManagers = new HashMap<>();
+    private String country;
+    private final ArrayList<String> countryParameters = new ArrayList<>();
+
+    private LocaleProvider localeProvider;
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        String weekEndDays = config.containsKey(CONFIG_WEEKENDAYS) ? config.get(CONFIG_WEEKENDAYS).toString() : null;
+        if (weekEndDays != null) {
+            String[] days = weekEndDays.split(",");
+            for (String day : days) {
+                weekend.add(DayOfWeek.valueOf(day.toUpperCase()));
+            }
+        } else {
+            logger.info("No days were configured as week-end days, you'll have to work all week");
+        }
+
+        country = config.containsKey(CONFIG_COUNTRY) ? config.get(CONFIG_COUNTRY).toString() : null;
+        if (country != null) {
+            logger.info("Default country '{}' overridden by configured one '{}'",
+                    localeProvider.getLocale().getCountry(), country);
+        } else {
+            country = localeProvider.getLocale().getCountry();
+            logger.info("Using system default country '{}' ", country);
+        }
+
+        if (config.containsKey(CONFIG_REGION)) {
+            countryParameters.add(config.get(CONFIG_REGION).toString());
+        }
+        if (config.containsKey(CONFIG_CITY)) {
+            countryParameters.add(config.get(CONFIG_CITY).toString());
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+    }
+
+    @Reference
+    protected void setLocaleProvider(LocaleProvider localeProvider) {
+        this.localeProvider = localeProvider;
+    }
+
+    protected void unsetLocaleProvider(LocaleProvider localeProvider) {
+        this.localeProvider = null;
+    }
+
+    @Override
+    public Collection<ParameterOption> getParameterOptions(URI uri, String param, Locale locale) {
+        if (CONFIG_URI.equals(uri.toString())) {
+            if (CONFIG_WEEKENDAYS.equals(param)) {
+                List<ParameterOption> options = new ArrayList<>();
+                for (DayOfWeek day : DayOfWeek.values()) {
+                    ParameterOption option = new ParameterOption(day.name(),
+                            day.getDisplayName(TextStyle.FULL, locale));
```

Comment:
While implementing changes on the VoiceManagerImpl it turned out that `locale` is null when configurations are saved. Therefore VoiceManagerImpl has a [`nullSafeLocale`](https://github.com/eclipse/smarthome/blob/de56ba83fe11287955cd8a305f9a33c9f987f0a5/bundles/core/org.eclipse.smarthome.core.voice/src/main/java/org/eclipse/smarthome/core/voice/internal/VoiceManagerImpl.java#L727) variable. It looks like the `DateTimeFormatterBuilder` can not handle null values for locale.